addons/hr_holidays/models/hr_leave.py

```diff
@@ -874,13 +943,61 @@ def add_follower(self, employee_id):
 
     @api.constrains('holiday_allocation_id')
     def _check_allocation_id(self):
+        allocations = self.env['hr.leave.allocation'].search([])
```

Comment:
This might return *a lot* of records, you should at least add a domain on the state, holiday_status_id, employee_id and ideally on the dates with `min`/`max`